### DISCUSSION TITLE
Question about js api usage

### DISCUSSION THREAD

**Discussion by dcaillibaud** [ruffle-rs/ruffle#1679] [Q&A] [answered]
Hi,

We have a lot of educational contents in flash (math interactive exercises for K12), and I tried to load them with ruffle but I've some pbs. Here are some questions (and not real issues)

1) Where to find js api documentation ? I read https://github.com/ruffle-rs/ruffle/wiki/Using-Ruffle but it's only basic. Is there any place to read https://github.com/ruffle-rs/js-docs online ? (to put http link in ours code comments)

2) Is there any way to pass flashvars (and other flash options) to ruffle using js api ? Eg for a swf which is a math animation editor I need to pass it a xml file with flashvars. I tried the classic swf embedding way then adding ruffle.js, passing swf?param=xxx to ruffle api, you can see some tests on https://instrumenpoche.sesamath.net/ruffle/ but it doesn't really work as expected :-/
In these tests, trying to load the pages with ruffle.js in chrome+ruffle extension crash ruffle ([test1](https://instrumenpoche.sesamath.net/ruffle/test1.html) and [test2](https://instrumenpoche.sesamath.net/ruffle/test2.html))

3) Is there any way to specify the wanted size for the canvas, I have big black margins on https://ressources.sesamath.dev/coll/lecteur/voir_flash_via_ruffle.php?dossier=am&fichier=762 and I don't find a way to solve this (it seems to be the unmuteOverlay in the shadow dom of the canvas)

Thanks a lot

**Comment by Dinnerbone** [comment, marked as answer]
Hey!

1. Our docs will be pushed to https://ruffle.rs/js-docs/master/index.html but they won't be accurate until tonights nightly, as it was dependant on a typescript rewrite landing (which did last night woo 🎉)
2. Either as part of the URL or currently as an extra parameter to the play methods - ie https://ruffle.rs/js-docs/master/classes/_ruffle_player_.ruffleplayer.html#streamswfurl - but I can see that it *is* working on your pages, I can see the request to `pentagone.xml` in network tools. I think the movie just isn't loading correctly in Ruffle yet, perhaps?
3. You can manually size the ruffle element for now (`player.style.width = "100px";` and such). We'll expose better controls over this, and a way to query the swfs preferred width/height, soon. The black bars are our letterboxing which we'll probably turn off.

**Comment by Poopooracoocoo** [comment]
@Dinnerbone (ermagerd I can't believe I'm mentioning dinnahbone) you should add a link to that website to the repo about section!

**Comment by Dinnerbone** [comment]
@Poopooracoocoo we absolutely will! As I said, the support for it only just landed

**Comment by dcaillibaud** [comment]
Thanks for this quick response (and all the ruffle work !)
1. :ok_hand: thanks
2. Nice, so the [actual way using js api](https://instrumenpoche.sesamath.net/ruffle/testApi.html) should work (`player.stream_swf_url('iep.swf?anim=pentagone.xml')`), it doesn't because other pbs (a lot in browser console), tested with firefox without flash neither ruffle extension (chrome with ruffle extension seems worse)
3. Perfect, I'll try with others swf

When I have loading pbs such as the link above, is it a good idea to open a new issue or should I have to wait actual issue tracking (avm1 for as2 support) to be reduced ?

**Comment by Dinnerbone** [comment]
Open a new issue, please! That way we have a nice concrete case of something that doesn't work, we can test with that and we can close it off and inform you when it does.

I'm going to turn this issue into a discussion as there was no actual issue here, just questions. Glad that my answers helped!
